watchdog/main.go

```diff
@@ -232,6 +232,34 @@ func getAdditionalEnvs(config *WatchdogConfig, r *http.Request, method string) [
 	return envs
 }
 
+func lockFilePresent() bool {
+	path := filepath.Join(os.TempDir(), ".lock")
+	if _, err := os.Stat(path); os.IsNotExist(err) {
+		return false
+	}
+	return true
+}
+
+func makeHealthHandler() func(http.ResponseWriter, *http.Request) {
```

Comment:
Can you explain what the health check is considering as healthy here? Can the presence of the lock file alone be considered sufficient to say that the function is healthy?

Reply:
According to issue , the /_/health endpoint need to check for `/tmp/.lock` file. 
https://github.com/openfaas/faas/issues/547

I think this will also make sure that function container has started successfully and functions can be invoked. Am I missing something?

Reply:
The lock file can be deleted at any time by the function to indicate an underlying problem.

Reply:
Ah ok.